server/sonar-web/src/main/js/apps/component-measures/sidebar/DomainFacet.js

```diff
@@ -59,26 +64,53 @@ export default class DomainFacet extends React.PureComponent {
     return measureSelected || overviewSelected;
   };
 
+  renderItemsFacet = () => {
+    const { domain, selected } = this.props;
+    const items = addMeasureCategories(domain.name, filterMeasures(domain.measures));
+    const hasCategories = items.some(item => typeof item === 'string');
+    const sortedItems = sortMeasures(domain.name, items);
+    return sortedItems.map(
+      item =>
+        typeof item === 'string'
+          ? <span key={item} className="facet search-navigator-facet facet-category">
+              <span className="facet-name">
+                {translate('component_measures.facet_category', item)}
+              </span>
+            </span>
+          : <FacetItem
+              active={item.metric.key === selected}
+              disabled={false}
+              key={item.metric.key}
+              name={
+                <span
+                  className={classNames({ 'big-spacer-left': hasCategories })}
+                  id={`measure-${item.metric.key}-name`}>
+                  {hasCategories
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3358)